gc3libs/workflow.py

```diff
@@ -401,7 +401,10 @@ def redo(self, from_stage=0, *args, **kwargs):
             task.redo(*args, **kwargs)
         # All other tasks should be put in NEW again
         for i in range(from_stage+1, len(self.tasks)):
-            self.tasks[i].execution.state = Run.State.NEW
+            try:
+                self.tasks[i].redo(*args, **kwargs)
+            except IndexError:
```

Comment:
I see the point of calling `.redo()` instead of resetting the state to `NEW` -- that's correct.

But why catching `IndexError`? The only expression I see that can raise `IndexError` is `self.tasks[i]` so then calling it in the `except:` clause would just raise `IndexError` again...